packages/py-moose-lib/moose_lib/streaming/streaming_function_runner.py

```
@@ -537,8 +543,10 @@ def process_messages():
                                 dlq,
                             ) in streaming_function_callables:
                                 try:
-                                    output_data = streaming_function_callable(
-                                        input_data
+                                    # Wrap user function execution with structured logging context
+                                    output_data = function_context.run(
+                                        {"function_name": log_prefix},
+                                        lambda: streaming_function_callable(input_data),
```

Ruff (0.14.14)

[warning] Function definition does not bind loop variable `streaming_function_callable`

(B023)